pyro/poutine/indep_messenger.py

```diff
@@ -59,7 +66,13 @@ class IndepMessenger(Messenger):
 
     """
 
-    def __init__(self, name=None, size=None, dim=None, device=None):
+    def __init__(
+        self,
+        name: str,
+        size: int,
+        dim: Optional[int] = None,
+        device: Optional[str] = None,
+    ):
```

Comment:
Here it doesn't make sense for `name` and `size` to be `None` (e.g. `self.size` is used to create `torch.arange` on line 137). So I decided to remove their default `None` values. Let me know if I should add them back.